src/NSwag.CodeGeneration.TypeScript/Templates/FetchClient.liquid

```diff
@@ -22,7 +22,7 @@
 {%     endif -%}
         this.http = http ? http : <any>window;
 {%     if UseGetBaseUrlMethod -%}
-        this.baseUrl = baseUrl ? baseUrl : this.getBaseUrl("{{ BaseUrl }}");
+        this.baseUrl = baseUrl  && baseUrl !== "" ? baseUrl : this.getBaseUrl("{{ BaseUrl }}");
```

Comment:
I'm a bit worried about this someone might want to provide "" as the value to use and not call getBaseUrl() in this case...

Maybe we should replace the whole line with

```
this.baseUrl = this.getBaseUrl("{{ BaseUrl }}", baseUrl);
```

So that the getBaseUrl() has the complete control over what baseUrl to use? 

This is a breaking change but gives much more flexibility and gives you a build error which can easily be fixed...

What do you think?

Reply:
Ok, I've changed the template.